tools/file_packager.py

```diff
@@ -192,15 +192,8 @@
   assert not separate_metadata, 'cannot separate-metadata without both --preloaded files and a specified --js-output'
 
 ret = '''
-var Module;
-'''
-if not no_closure:
-  ret += '''
-if (typeof Module === 'undefined') Module = eval('(function() { try { return Module || {} } catch(e) { return {} } })()');
-'''
-else:
-  ret += '''
-if (typeof Module === 'undefined') Module = {};
+// `"__EMSCRIPTEN_PRIVATE_MODULE_EXPORT_NAME_SUBSTITUTION__"` will be replaced with actual code on later stage of the build, this way Closure Compiler will not mangle it
```

Comment:
i'm not sure we can do this in the file packager. file packager output may be combined with the compiler JS and optimized by closure, but it also might not - you can load file packager output as a standalone file, after an emscripten program is running, and it will import files into the filesystem.

Reply:
Does this mean `module_export_name_substitution()` will not be executed on compiled output in that case? Because `module_export_name_substitution()` is executed independently from `--closure` option.

Reply:
Yes. I should have noticed this issue earlier, sorry.

Basically, you can build one JS executable, and run the file packager yourself - without emcc - multiple times to create a bunch of other JS files. emcc will never see those, but they can be combined with the JS executable from emcc later.

So there are two use cases for the file packager, when it is called from emcc - in which case the output will be combined and optimized with the JS from emcc - and when it is called standalone.

When emcc invokes the file packager, it can tell it what to do for closure support. Right now we actually have a flag, `--no-closure` which it can pass. Probably it should also forward the EXPORT_NAME if there is one, which would fix #5089 that was just filed. Is that enough for this too?

Reply:
Actually I don't really know what packager is, why it exists in the first place and why some code is duplicated from other places here.

Reply:
Thanks for detailed explanation. Didn't understand the use case entirely, but got enough to implement what you're requesting. Will work on it.